packages/plugin-sql/src/base.ts

```
@@ -1275,6 +1282,48 @@ export abstract class BaseDrizzleAdapter<
+        .replace(/\\u(?![0-9a-fA-F]{4})/g, '\\\\u'); // Fix malformed Unicode escape sequences
+    }
+
+    if (typeof value === 'object') {
+      if (Array.isArray(value)) {
+        return value.map(item => this.sanitizeJsonObject(item));
+      } else {
+        const result: Record<string, unknown> = {};
+        for (const [key, val] of Object.entries(value)) {
+          // Also sanitize object keys
+          const sanitizedKey = typeof key === 'string' ? 
+            key.replace(/\u0000/g, '').replace(/\\u(?![0-9a-fA-F]{4})/g, '\\\\u') : 
```

Biome (1.9.4)

[error] Unexpected control character in a regular expression.

Control characters are unusual and potentially incorrect inputs, so they are disallowed.

(lint/suspicious/noControlCharactersInRegex)